rio_tiler/io/xarray.py

```diff
@@ -71,7 +82,20 @@ def __attrs_post_init__(self):
         assert rioxarray is not None, "rioxarray must be installed to use XarrayReader"
 
         self.bounds = tuple(self.input.rio.bounds())
+
+        if not self.input.rio.crs and default_to_wgs84_crs:
+            warnings.warn(
+                "Dataset doesn't have a valid CRS set. Automatically setting CRS to WGS84",
+                OverwritingCRSWarning,
+            )
+            self.input.rio.write_crs("epsg:4326", inplace=True)
```

Comment:
This mutates the `self.input` Xarray object (DataArray).

If we do not copy what the user passed in earlier, this behaviour could be unexpected for a user.

If there is a chance that the XarrayReader mutates the input, should we perhaps always have it create a copy up-front?

Reply:
🤔 good point

Reply:
😬 sadly rio-tiler gets the input directly from the user so we can't really avoid `editing` the origin data array